aten/src/ATen/native/ComplexHelper.h

```diff
@@ -48,6 +48,7 @@ Tensor view_as_complex(const Tensor& self) {
     self.scalar_type() == kFloat || self.scalar_type() == kDouble || self.scalar_type() == kHalf,
     "view_as_complex is only supported for half, float and double tensors, but got a tensor of scalar type: ", self.scalar_type());
 
+  TORCH_CHECK(self.dim() != 0, "Dimension of tensor can't be zero");
```

Comment:
maybe - `"Input tensor must have one or more dimensions."`

Reply:
alternately the check below can also be just modified to:
`TORCH_CHECK(self.dim() != 0 && new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");`